projects/gateway2/translator/plugins/virtualhostoptions/virtualhost_options_plugin.go

```diff
@@ -174,18 +178,43 @@ func (p *plugin) InitStatusPlugin(ctx context.Context, statusCtx *plugins.Status
 		// for this specific proxy, get all the route errors and their associated RouteOption sources
 		virtualHostErrors := extractVirtualHostErrors(proxyWithReport.Reports.ProxyReport)
 		for vhKey := range virtualHostErrors {
-			cacheEntry := &classicStatus{
-				subresourceStatus: map[string]*core.Status{},
-				virtualHostErrors: []*validation.VirtualHostReport_Error{},
-				warnings:          []string{},
-			}
+			cacheEntry := newClassicStatus()
 			// init the cache
 			p.classicStatusCache[vhKey] = cacheEntry
 		}
 	}
 	return nil
 }
 
+// MergeStatusPlugin merges the status of the source plugin in to the current plugin.
+// This is used late in the proxy sync process to report the status of the VHO when
+// it references multiple proxies/snapshots.
+func (p *plugin) MergeStatusPlugin(ctx context.Context, source any) error {
```

Comment:
would it be possible to avoid this method altogether by merging the StatusContext?

Reply:
Each plugin may have it's own internal state/struct that need plugin-specific logic. I wanted to make sure to support that that case, just like we do when applying the statuses.